build.gradle

```diff
@@ -90,6 +97,9 @@ task jextractapp() {
     dependsOn copyLibClang
 }
 
+// build the jextract image when the build or assemble task is run
+assemble.dependsOn(jextractapp)
+
```

Comment:
Not sure of this change. Basically what we want is only three targets to use "clean", "verify" and "jtreg"

"verify" is for just create minimal jextract JDK image and run simple 'integration test' on it.
'jtreg' is for really running all tests

Note that we cannot use usual lifecycle of Gradle projects. Out tests are run a separate JDK image containing jextract tool in it - because this is how we expect users to deploy/use.

"verify" creates a minimal jextract image (which won't have compiler and other things needed for jtreg test run and so we create two JRE images) and runs a simple jextract on a simple test.h.

Reply:
"jtreg" target requires additional toolset (cmake, C compiler, jtreg) at the user end. This is not the usual pure Java project "tests" which don't need any of these. Already "verify" doesn't depend on "jtreg" (verify is just build of jextract image + a simple integration test to extract a test.h). "jtreg" perhaps can skip the "project jextract image" + "integration test" step. i.e., after jar straight build test JDK image with jextract and run all jextract tests. But, not sure how much is saved by that.

Reply:
To me at least, it would make sense for the "build" task to create the final artifact designed for users. This help would avoid confusion from people who didn't fully read the README, and try to simply run the "build" task (then try to run the jar in build/libs or otherwise do something incorrect/unsupported). Perhaps it may even be best to print a message announcing the correct location of the final artifact since developers familiar with gradle may assume that it just builds a executable jar like most projects. However I also understand how this project doesn't fully fit into the gradle build lifecycle. I'll leave it up to you two to decide what we want the final build process to look like. I'm happy to implement whatever approach you decide on in this PR.

Reply:
At the moment it is:

build -> build the jextract classes
verify -> build jextract image and run small integration test - depends on build
jtreg -> run all jtreg tests - depends on verify

Now, we don't want "jtreg" be part of the build pipeline, because running tests require setting up compilation of native libraries, which might not be possible for users. 

That said, on "verify" we're more flexible. I think another good stacking could be:

build -> build the jextract classes and image
verify -> run small integration test - depends on build
jtreg -> run all jtreg tests - depends on verify

This means that doing `gradle build` (which is probably what majority of users will reach out for first) will produce something useful.

Reply:
This change currently implements the second stacking you propose (with the minor detail of assemble depending on jextractapp since build depends on assemble by default). Is this acceptable or should I revert this change?

Reply:
I've tried out this patch, and I confirm that it behaves as described (e.g. second stacking). I had an offline discussion with @sundararajana who is also ok with this change. This can go ahead IMHO. I will approve.